wordpress-mobile/WordPress-Android
# Move some secrets outside the repository

Moves the low-hanging fruits in terms of secrets outside of the repository:

- Fastlane's `google-upload-cerdentials.json`
- The debug keystore

## To test

Checkout this branch and run `bundle exec fastlane run configure_apply`, then `ls ~/.configure/wordpress-android/secrets`, you should see:

```
➜ ls ~/.configure/wordpress-android/secrets

debug_a8c.keystore             google-upload-credentials.json
```

**Individual tests**

`debug_a8c.keystore`: As far as I can see, the debug keystore is only used in CI, in particular in CircleCI? I run all the CircleCI steps and they worked. The fact that we don't need to worry about that keystore for local development seem to be confirmed by the [_Sign your debug build_](https://developer.android.com/studio/publish/app-signing#debug-mode) documentation.

`google-upload-credentials.json`: This file is used by the `upload_to_play_store` Fastlane action. I checked the action [documentation](https://docs.fastlane.tools/actions/upload_to_play_store/) and couldn't find a way to perform a dry-run. I saw there is a `validate_only` option, but the documentation says "Only validate changes with Google Play rather than actually publish," which doesn't make it clear to me whether data would be uploaded or not. I'm not familiar with the upload steps in this repo and would like to avoid uploading stuff accidentally.

@AliSoftware and @oguzkocer, I think you are the most knowledgeable when it comes to this part of the WordPress Android codebase: any suggestion on how to further verify these changes before merging? Thanks!

## Regression Notes
1. Potential unintended areas of impact

I don't think these changes may impact areas other than what specified above, but I'll defer to those who have more know-how in Android and this repo.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N.A.

3. What automated tests I added (or what prevented me from doing so)
N.A.

---
